bin/cron/sync_contact_rollups

```diff
@@ -0,0 +1,123 @@
+  contact_rollups_src = PEGASUS_REPORTING_DB_READER['SELECT * FROM contact_rollups_daily FORCE INDEX(contact_rollups_email_index) ORDER BY email']
+  # Query all of the contacts in the master contact rollup table (contact_rollups_daily) sorted by email.
+  contact_rollups_dest = PEGASUS_DB_READER['SELECT * FROM contact_rollups FORCE INDEX(contact_rollups_email_index) ORDER BY email']
+
+  # Create iterators for both queries using the #stream method so we stream the results back rather than
+  # trying to load everything in memory
+  src_iterator = contact_rollups_src.stream.to_enum
+  dest_iterator = contact_rollups_dest.stream.to_enum
+
+  # Do a row-by-row comparison of the new daily contact rollup (src) against the existing (dest) and calculate
+  # the differences. Make inserts or updates to make dest match src. Wherever we do an insert or an update,
+  # mark the updated_at timestamp so we track what rows have changed at what time, to enable efficient sync
+  # into Pardot.
+
+  contact_rollup_src = grab_next(src_iterator)
+  contact_rollup_dest = grab_next(dest_iterator)
+  until contact_rollup_src.nil?
+
+    email_src = contact_rollup_src[:email]
+
+    # Continue to advance the destination pointer until the destination email address in question
+    # is the same or later alphabetically as the source email address.
+    while (!contact_rollup_dest.nil?) &&
+        ((contact_rollup_dest[:email] <=> email_src) == -1)
+      contact_rollup_dest = grab_next(dest_iterator)
+    end
+
+    # Determine if this is a new record (email address doesn't exist in destination table) or existing.
+    is_new = true
+    unless contact_rollup_dest.nil?
+      email_dest = contact_rollup_dest[:email]
+      # Compare email addresses of source and destination row to see if they are the same.
+      # IF not, the source record is new.
+      is_new = (email_src.casecmp(email_dest) != 0)
+    end
+
+    # Build the set of column data that needs to be inserted/updated
+    output_row = {}
+    contact_rollup_src.each do |column, value|
+      # Skip columns that include sync metadata
+      next if EXCLUDED_COLUMNS.include? column.to_s
+
+      # If this is not a new record, skip columns where the data already matches
+      next if !is_new && (value == contact_rollup_dest[column])
+
+      # Add the source data to the same column in the destination row
+      output_row[column] = value
+    end
+
+    # If there are no columns to update, then this is an existing record where the
+    # destination is the same as the source, nothing to do
+    if output_row.empty?
+      num_unchanged += 1
+    else
+
+      # Track the timestamp of the change, so the sync process knows this change needs to be synced
+      output_row[:updated_at] = Time.now
+
+      if is_new
+        # Insert the destination record
+        log("#{Time.now} Inserted #{email_src} (src id: #{contact_rollup_src[:id]})")
```

Comment:
Not important, but if shakedown runs are meant to be manual runs instead of cron, then you could enable it with a `--verbose` command line option or an environment variable (save commenting / uncommenting the code, and save the cronjob from dumping tons of logs).